DigitalLearningSolutions.Data/Helpers/FilteringHelper.cs

```diff
@@ -122,6 +129,117 @@ FilterOptions filterOptions
+        }
+
+        public static string GetFilterValueForRegistrationPrompt(int promptNumber, string? answer, string prompt)
+        {
+            var filterValueName = GetFilterValueForRegistrationPrompt(promptNumber, prompt);
+            var propertyValue = string.IsNullOrEmpty(answer)
+                ? EmptyValue
+                : answer;
+            return BuildFilterValueString(filterValueName, filterValueName, propertyValue);
+        }
+
+        private static IEnumerable<FilterOptionModel> GetFilterOptionsForPromptWithOptions(Prompt prompt)
+        {
+            var filterValueName = prompt is CentreRegistrationPrompt registrationPrompt
+                ? GetFilterValueForRegistrationPrompt(registrationPrompt.RegistrationField.Id, prompt.PromptText)
+                : GetFilterValueForAdminField(((CourseAdminField)prompt).PromptNumber, prompt.PromptText);
+
+            var options = prompt.Options.Select(
+                option => new FilterOptionModel(
+                    option,
+                    BuildFilterValueString(filterValueName, filterValueName, option),
+                    FilterStatus.Default
+                )
+            ).ToList();
+            options.Add(
+                new FilterOptionModel(
+                    "No option selected",
+                    BuildFilterValueString(
+                        filterValueName,
+                        filterValueName,
+                        EmptyValue
+                    ),
+                    FilterStatus.Default
+                )
+            );
+            return options;
+        }
+
+        private static IEnumerable<FilterOptionModel> GetFilterOptionsForPromptWithoutOptions(Prompt prompt)
+        {
+            var filterValueName = prompt is CentreRegistrationPrompt registrationPrompt
+                ? GetFilterValueForRegistrationPrompt(registrationPrompt.RegistrationField.Id, prompt.PromptText)
+                : GetFilterValueForAdminField(((CourseAdminField)prompt).PromptNumber, prompt.PromptText);
+
+            var options = new List<FilterOptionModel>
+            {
+                new FilterOptionModel(
+                    "Not blank",
+                    BuildFilterValueString(
+                        filterValueName,
+                        filterValueName,
+                        FreeTextNotBlankValue
+                    ),
+                    FilterStatus.Default
+                ),
+                new FilterOptionModel(
+                    "Blank",
+                    BuildFilterValueString(
+                        filterValueName,
+                        filterValueName,
+                        FreeTextBlankValue
+                    ),
+                    FilterStatus.Default
+                ),
+            };
+            return options;
+        }
+
+        private static string GetFilterValueForRegistrationPrompt(int promptNumber, string promptText)

```

Comment:
This is really a filter name / key right? I'd have thought a filter value was the prompt answer you're filtering by. Same with admin field method.

Reply:
With the renaming done in the comment below, I think this is suitable.